### PR TITLE
feat: add optional Access-Control-Max-Age header to CORS plugin

### PR DESCRIPTION
**Pull request by osamra-rbi**
Hi there

As discussed in this [issue](https://github.com/apollographql/router/issues/2212), the CORS plugin in Apollo Router currently lacks setting the Access-Control-Max-Age header. This PR adds a new option called max_age that is used like so:
```
cors:
  max_age: 86400
```

This does not solve the Cache-Control header mentioned in issue as it's a bit separate from CORS.

New test(s) and documentation changes haven't been done yet as I wanted to confirm this is fine before proceeding further. If so, happy to do them. Thanks!

(FYI I don't have the ability to assign a reviewer or anything, if we're able to do that)

### CODE REVIEW

**Comment by SimonSapin** [comment]
I’ve pushed an update of test expectation.

@Geal where would be the right place to document this new feature?

File: apollo-router/src/configuration/snapshots/apollo_router__configuration__tests__schema_generation.snap [hunk 1]
```diff
@@ -63,6 +64,14 @@ expression: "&schema"
           },
           "nullable": true
         },
+        "max_age": {
+          "description": "The `Access-Control-Max-Age` header value in seconds",
+          "default": null,
+          "type": "integer",
```
Comment (abernix) [inline apollo-router/src/configuration/snapshots/apollo_router__configuration__tests__schema_generation.snap:70]: I think elsewhere in our configuration, the pattern is for this to accept a human-friendly value that more clearly lets the reader understand the unit of measurement: e.g., accepting `5m` for 5 minutes or `3600s` for 3600 seconds.

Choosing to merely accept seconds as an integer would deviate from that convention.  WDYT?

Comment (bnjjj) [reply apollo-router/src/configuration/snapshots/apollo_router__configuration__tests__schema_generation.snap:70]: Example here https://github.com/apollographql/router/blob/bnjjj/fix_2406/apollo-router/src/plugins/authentication.rs#L80:L80

Comment (osamra-rbi) [reply apollo-router/src/configuration/snapshots/apollo_router__configuration__tests__schema_generation.snap:70]: I think that's fair, I can take a shot at this and also add the necessary documentation

Comment (osamra-rbi) [reply apollo-router/src/configuration/snapshots/apollo_router__configuration__tests__schema_generation.snap]: Addressed!

**Review by bnjjj — APPROVED** [review]

**Comment by bnjjj** [comment]
@osamra-rbi could run a `cargo xtask test` locally to update test snapshots please ?

**Comment by osamra-rbi** [comment]
> @osamra-rbi could run a `cargo xtask test` locally to update test snapshots please ?

Woops, sorry, wasn't sure how that worked. Done! Should I add a test to tests.rs as well?

**Comment by bnjjj** [comment]
@osamra-rbi That would be great to add a test if you can yes !

**Comment by bnjjj** [comment]
@osamra-rbi just small lints to fix https://app.circleci.com/pipelines/github/apollographql/router/9616/workflows/5dd1863d-36d2-451e-bcf6-22a692adba8e/jobs/66125 :)

**Comment by bnjjj** [comment]
I just added a changelog entry.  I enable auto merge to merge it as soon as all checks are green :)

**Pull request merged**